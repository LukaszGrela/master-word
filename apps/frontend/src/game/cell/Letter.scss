@import '../../styles/variables';

.letter {
  height: var(--gd-board-size);
  box-sizing: border-box;
  background-color: $letter;
  display: flex;
  flex-direction: column;
  justify-content: center;

  position: relative;
  pointer-events: none;
  & .container {
    font-weight: 500;
    font-size: 1.5rem;
    user-select: none;
  }

  & .icon {
    display: none;
  }

  & .flag {
    display: none;
    &-wrapper {
      display: inline-flex;
    }

    & img {
      width: 1rem;
    }
  }

  &.inactive {
    background-color: $letter-inactive;
    &.correct {
      background-color: $letter-correct;
      color: $tintedWhite;
    }

    &.misplaced {
      background-color: $letter-misplaced;
    }

    &.incorrect {
      background-color: $letter-incorrect;
    }
  }

  &.active {
    background-color: $letter-active;
    color: $letter-active-color;
    &.wrong {
      border-color: $letter-wrong;
      border-top: 1px solid $letter-wrong;
      border-bottom: 2px solid $letter-wrong;

      animation: blink 0.75s;
      animation-iteration-count: 3;
    }
    &.wrong.first {
      border-left: 1px solid $letter-wrong;
    }
    &.wrong.last {
      border-right: 1px solid $letter-wrong;
    }

    &.first .flag {
      display: inline-flex;
      align-items: center;

      top: 1.5rem;
      left: -0.55rem;
      position: absolute;
      background-color: inherit;
      padding: 0.3rem 0.3rem;
      border-radius: 1rem;

      .mobile &,
      .mobile & {
        top: 2rem;
        @media screen and (max-width: 320px) {
          bottom: 0;
          border: none;
          padding: 0;
          background: transparent;
          left: 2px;
        }
      }
    }

    .mobile &.show-icon .icon {
      display: block;
    }
  }
  .mobile & .icon svg {
    fill: #00000057;
    position: absolute;
    left: 0;
    top: 0;
    width: 1.75rem;
    height: 1.75rem;
  }
}

@keyframes blink {
  50% {
    border-color: transparent;
  }
}

@media (prefers-color-scheme: dark) {
  .letter {
    background-color: $letter-dark;
  }
  .letter.inactive {
    background-color: $letter-inactive-dark;
  }
  .letter.inactive.misplaced {
    background-color: $letter-misplaced-dark;
  }
  .letter.active {
    background-color: $letter-active-dark;
    color: $letter-active-color-dark;
  }
}
