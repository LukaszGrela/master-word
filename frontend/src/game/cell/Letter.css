.letter {
  height: 4rem;
  box-sizing: border-box;
  background-color: seashell;
  display: flex;
  flex-direction: column;
  justify-content: center;

  position: relative;
  pointer-events: none;
}

.letter .container {
  font-weight: 500;
  font-size: 1.5rem;
  user-select: none;
}

.letter .icon {
  display: none;
}

.mobile .letter.active.show-icon .icon {
  display: block;
}

.mobile .letter .icon svg {
  fill: #00000057;
  position: absolute;
  left: 0;
  top: 0;
  width: 2rem;
  height: 2rem;
}

.letter.inactive {
  background-color: hsl(25 40% 90% / 1);
}

.letter.active {
  background-color: hsl(25 100% 91% / 1);
}

.letter.inactive.correct {
  background-color: green;
  color: rgba(255, 255, 255, 0.87);
}

.letter.inactive.misplaced {
  background-color: orange;
}

.letter.incorrect.inactive {
  background-color: grey;
}

.letter.active.wrong {
  border-color: red;
  border-top: 1px solid red;
  border-bottom: 2px solid red;

  animation: blink 0.75s;
  animation-iteration-count: 3;
}
.letter.active.wrong.first {
  border-left: 1px solid red;
}
.letter.active.wrong.last {
  border-right: 1px solid red;
}

@keyframes blink {
  50% {
    border-color: transparent;
  }
}
