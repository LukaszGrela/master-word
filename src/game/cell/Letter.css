.letter {
  height: 4rem;
  box-sizing: border-box;
  background-color: seashell;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.letter .container {
  font-weight: 500;
  font-size: 1.5rem;
  user-select: none;
}

.letter.inactive {
  background-color: hsl(25 40% 90% / 1);
}

.letter.active {
  background-color: hsl(25 100% 91% / 1);
}

.letter.inactive.correct {
  background-color: green;
  color: rgba(255, 255, 255, 0.87);
}

.letter.inactive.misplaced {
  background-color: orange;
}

.letter.incorrect.inactive {
  background-color: grey;
}
