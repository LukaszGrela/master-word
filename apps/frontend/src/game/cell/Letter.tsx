import React from 'react';
import { classNames } from '../../utils/classNames';
import { ILetter } from './types';
import { EditIcon } from '../icons/EditIcon';
import './Letter.scss';

const Letter: React.FC<ILetter> = ({ className, letter }): JSX.Element => {
  const cn = classNames('letter', className);
  return (
    <div className={cn}>
      <span className="icon">
        <EditIcon />
      </span>
      <span className="container">{letter}</span>
    </div>
  );
};

export default Letter;
