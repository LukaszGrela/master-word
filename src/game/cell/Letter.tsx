import React from 'react';
import { classNames } from '../../utils/classNames';
import { ILetter } from './types';
import './Letter.css';
import { EditIcon } from '../icons/EditIcon';

const Letter: React.FC<ILetter> = ({ className, letter }): JSX.Element => {
  const cn = classNames('letter', className);
  return (
    <div className={cn}>
      <span className='icon'>
        <EditIcon />
      </span>
      <span className='container'>{letter}</span>
    </div>
  );
};

export default Letter;
