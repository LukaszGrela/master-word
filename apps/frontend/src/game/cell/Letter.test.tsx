import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';

import Letter from './Letter';

describe('game', () => {
  describe('cell', () => {
    describe('Letter', () => {
      it('Renders with no props', () => {
        const { container } = render(<Letter />);

        expect(container).toMatchSnapshot();
      });
      it('Renders with all props', () => {
        const { container } = render(
          <Letter className="test" language="en" letter="E" />,
        );

        expect(container).toMatchSnapshot();
      });
    });
  });
});
