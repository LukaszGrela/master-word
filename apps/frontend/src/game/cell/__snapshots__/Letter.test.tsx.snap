// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`game > cell > Letter > Renders with all props 1`] = `
<div>
  <div
    class="letter test"
  >
    <span
      class="flag"
    >
      <span
        class="flag-wrapper"
      >
        <img
          alt=""
          loading="lazy"
          src="https://flagcdn.com/w20/us.png"
          srcset="https://flagcdn.com/w40/us.png 2x"
          width="20"
        />
      </span>
    </span>
    <span
      class="icon"
    >
      <svg
        class="edit-icon"
        height="48"
        viewBox="0 -960 960 960"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M167-120q-21 5-36.5-10.5T120-167l40-191 198 198-191 40Zm191-40L160-358l473-473q17-17 42-17t42 17l114 114q17 17 17 42t-17 42L358-160Zm317-628L233-346l113 113 442-442-113-113Z"
        />
      </svg>
    </span>
    <span
      class="container"
    >
      E
    </span>
  </div>
</div>
`;

exports[`game > cell > Letter > Renders with no props 1`] = `
<div>
  <div
    class="letter"
  >
    <span
      class="flag"
    >
      <span
        class="flag-wrapper"
      >
        <img
          alt=""
          loading="lazy"
          src="https://flagcdn.com/w20/pl.png"
          srcset="https://flagcdn.com/w40/pl.png 2x"
          width="20"
        />
      </span>
    </span>
    <span
      class="icon"
    >
      <svg
        class="edit-icon"
        height="48"
        viewBox="0 -960 960 960"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M167-120q-21 5-36.5-10.5T120-167l40-191 198 198-191 40Zm191-40L160-358l473-473q17-17 42-17t42 17l114 114q17 17 17 42t-17 42L358-160Zm317-628L233-346l113 113 442-442-113-113Z"
        />
      </svg>
    </span>
    <span
      class="container"
    />
  </div>
</div>
`;
